src/main/java/io/tiledb/spark/TileDBDataWriter.java

```diff
@@ -456,25 +459,43 @@ public void write(InternalRow record) throws IOException {
 
   private void flushBuffers() throws TileDBError {
     this.metricsUpdater.startTimer(queryWriteFlushBuffersTimerName);
+    JavaArray jArray;
     long buffersInBytes = 0;
-    query.setBuffer(Constants.TILEDB_COORDS, nativeArrayBuffers[0], nRecordsBuffered * nDims);
+
+    query.setBuffer(
+        Constants.TILEDB_COORDS,
+        new NativeArray(
+            ctx,
+            javaArrayBuffers[0].get(),
+            javaArrayBuffers[0].getDataType(),
+            javaArrayBuffers[0].getNumElements()),
+        nRecordsBuffered * nDims);
     // Calculate bytes we are writing for metrics starting with dimension
     buffersInBytes += nRecordsBuffered * nDims * bufferDatatypes[nDims - 1].getNativeSize();
     for (int i = nDims; i < bufferNames.length; i++) {
```

Comment:
I know this is existing code but is this right? `i = nDims`? There is only one buffer for coordinates and that is at position 0 of javaArrayBuffers right? So should this be hard coded to start at 1?

Reply:
No this is correct thanks to @gsvic 's analysis.